pkg/syncer/pvcsi_fullsync.go

```diff
@@ -431,7 +431,7 @@ func setGuestClusterDetailsOnSupervisorPVC(ctx context.Context, metadataSyncer *
 			// Add label to Supervisor PVC that contains guest cluster details, if not present already.
 			key := fmt.Sprintf("%s/%s", metadataSyncer.configInfo.Cfg.GC.TanzuKubernetesClusterName,
 				metadataSyncer.configInfo.Cfg.GC.ClusterDistribution)
-			if _, ok := svPVC.Labels[key]; !ok {
+			if val, ok := svPVC.Labels[key]; !ok || val != metadataSyncer.configInfo.Cfg.GC.TanzuKubernetesClusterUID {
```

Comment:
can we make similar change in setGuestClusterDetailsOnSupervisorSnapshot() where we add the label to snapshot ?

Reply:
done